.packit.yaml

```diff
@@ -0,0 +1,58 @@
+# This file is used by the Packit service.
+# Currently it is used for the gating of pull requests.
+#
+# See the documentation for more information:
+# https://packit.dev/docs/configuration/
+
+# We need a specfile to build the RPM package.
+# But instead of storing it in this repository,
+# we use the one that exists already in src.fedoraproject.org/rpms/golang.
+# In particular, we use the specfile from the branch `eln`.
+# We'll create the temporary folder .packit_rpm to store the content of the
+# Fedora repository.
+specfile_path: .packit_rpm/golang.spec
+
+# We need to tell Packit which files to sync from the upstream repository.
+files_to_sync:
+  - .packit.yaml
+  - ./scripts/packit.sh
+  - src: .packit_rpm/golang.spec
+    dest: golang.spec
+
+srpm_build_deps:
+  - golang
+  - net-tools
+  - openssl-devel
+  - glibc-static
+  - perl-interpreter
+  - procps-ng
+
+# The name of the package in the upstream repository.
+upstream_package_name: golang
+# The name of the package in Fedora.
+downstream_package_name: golang
+# Use a different tag template for the upstream repository.
+#upstream_tag_template: "v{version}"
+
+actions:
+  create-archive:
+    - "bash ./scripts/packit.sh create-archive"
+  post-upstream-clone:
+    # Use the Fedora ELN package.
+    - "git clone https://src.fedoraproject.org/rpms/golang.git .packit_rpm --branch eln"
+  fix-spec-file:
+    # Fix the specfile by running the ./scripts/packit.sh script
+    # We cannot put the content of the script inside the yaml due to limitations of the format
+    - "bash ./scripts/packit.sh"
+
+jobs:
+    # Build the package in Fedora ELN when a pull request is opened.
+    # It can be requested by adding a comment with the text: /packit copr-build
+  - job: copr_build
+    trigger: pull_request
+    targets:
+    - fedora-eln-aarch64
+    - fedora-eln-ppc64le
+    - fedora-eln-s390x
+    - fedora-eln-x86_64
```

Comment:
quick note, currently we don't have epel-10 targets on copr, so it'll only be epel-9. But centos stream can have both 9 and 10.

Reply:
I am concern that will increase the complexity as CentOS Stream and ELN spec files might be potentially different at some times, but we should try on CentOS Stream too for sure.

Not sure about EPEL, why is necessary? There is no golang spec file for EPEL 8/9.

Reply:
> I am concern that will increase the complexity as CentOS Stream and ELN spec files might be potentially different at some times, but we should try on CentOS Stream too for sure.

There are maybe two approaches you can try:
1. Make both ELN and CentOS Stream 9/10 buildable using a single spec file.
2. Include separate spec files for both ELN and CentOS Stream and mention them like [this](https://github.com/containers/container-selinux/blob/main/.packit.yaml#L23-L29) . You can replace you existing `specfile_path` line with that and use the separate paths for each of fedora and centos-stream.

I guess we could leave that for a followup PR.

> 
> Not sure about EPEL, why is necessary? There is no golang spec file for EPEL 8/9.

This could instead be a `rhel-9-$arch` copr target as well.

BTW, a quirk about copr is that on a rhel / centos-stream env if you run `dnf copr enable foo/bar` it will by default enable the `epel` target if it exists. And the epel copr targets are `rhel + epel` envs. So, if you wanna check buildability on proper rhel, you could use either of `rhel-` or `epel-` and get the same results. Just that the copr  quirk might make using epel a little more convenient for testing etc.

This could also be left for a followup.